### DISCUSSION TITLE
Updating metadata independently of CsWin32 breaks the projection

### DISCUSSION THREAD

**Discussion by AliveDevil** [microsoft/CsWin32#166] [Q&A] [answered]
So … after Win32Metadata preview 26 CsWin32 does not generate GUID headers anymore, due to a change in how GUID attributes are specified in metadata:
Until Preview-22:
```
[Guid("4657278B-411B-11D2-839A-00C04FD918D0")]
```
After Preview-22:
```
[Windows.Win32.Interop.Guid(1180116875u, 16667, 4562, 131, 154, 0, 192, 79, 217, 24, 208)]
```

I think this might break the SpecllChecker-sample as well.

**Comment by AArnott** [comment, marked as answer]
Correct. Each new metadata update has a high likelihood of breaking the projections that consume it because breaking changes such as this one are still quite common as we're in early preview. In general it's a safer plan to just wait for the CsWin32 update that brings the updated metadata with it rather than try to update the metadata directly.
Of course we specifically did work to enable folks to update the metadata independently so that folks can do what you just did. It just isn't practical to rely on it always working yet.
